### PR TITLE
feat(zk_toolbox): Make chain init independent from ecosystem

### PR DESCRIPTION
**Pull request by matias-gonz**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: zkstack_cli/crates/zkstack/src/commands/chain/args/init/mod.rs [hunk 1]
```diff
@@ -80,14 +92,40 @@ impl InitArgs {
...
+        };
+
+        let wallets_path = ecosystem.map_or_else(
+            || {
+                self.wallets_path
+                    .map_or_else(|| Prompt::new(MSG_WALLETS_PATH_PROMPT).ask(), PathBuf::from)
+            },
+            |e| e.get_wallets_path(),
+        );
+
```
Comment (Deniallugo) [inline zkstack_cli/crates/zkstack/src/commands/chain/args/init/mod.rs:118]: from messages it's not clear what contracts and wallets you want

Comment (matias-gonz) [reply zkstack_cli/crates/zkstack/src/commands/chain/args/init/mod.rs:118]: I updated the wallets message in: [f194e18](https://github.com/matter-labs/zksync-era/pull/3231/commits/f194e18734715dd47ccbe149c70611a53e6afa71)

Is this OK or what should the message be?

Comment (matias-gonz) [reply zkstack_cli/crates/zkstack/src/commands/chain/args/init/mod.rs:118]: Prompt for contracts path is:
`Provide the path to the ecosystem contracts or keep it empty and you will use ZKsync ecosystem config. For using this config, you need to have governance wallet`
How can we improve it?

File: zkstack_cli/crates/zkstack/src/commands/chain/args/init/configs.rs [hunk 2]
```diff
@@ -51,12 +64,43 @@ impl InitConfigsArgs {
...
+}
+
+pub fn get_ecosystem_contracts_path(
+    ecosystem_contracts_path: Option<String>,
+    ecosystem: Option<EcosystemConfig>,
+    chain: &ChainConfig,
+) -> anyhow::Result<PathBuf> {
+    let ecosystem_preexisting_configs_path = ecosystem.map_or_else(
+        || chain.get_preexisting_ecosystem_contracts_path(),
+        |e| e.get_preexisting_ecosystem_contracts_path(),
```
[the remark this answers is not in the document]
Comment (matias-gonz) [reply zkstack_cli/crates/zkstack/src/commands/chain/args/init/configs.rs:86]: Fixed!
[537a33b](https://github.com/matter-labs/zksync-era/pull/3231/commits/537a33b99853106e4fcd5d65a1153b9363f452c1)

**Pull request merged**